public/app/features/alerting/unified/components/contact-points/utils.ts

```diff
@@ -87,38 +91,50 @@ export interface ReceiverConfigWithMetadata extends GrafanaManagedReceiverConfig
 }
 
 export interface ContactPointWithMetadata extends GrafanaManagedContactPoint {
-  numberOfPolicies: number;
+  numberOfPolicies?: number; // now is optional as we don't have the data from the read-only endpoint
   grafana_managed_receiver_configs: ReceiverConfigWithMetadata[];
 }
 
 /**
  * This function adds the status information for each of the integrations (contact point types) in a contact point
  * 1. we iterate over all contact points
  * 2. for each contact point we "enhance" it with the status or "undefined" for vanilla Alertmanager
+ * resultFromConfig: is passed when we need to get number of policies for each contact point
+ * resultFromReadOnlyEndpoint: is passed when there is no need to get number of policies for each contact point,
+ * and we prefer using the data from the read-only endpoint.
  */
 export function enhanceContactPointsWithMetadata(
-  result: AlertManagerCortexConfig,
   status: ReceiversStateDTO[] = [],
   notifiers: NotifierDTO[] = [],
-  onCallIntegrations: OnCallIntegrationDTO[] | undefined | null
+  onCallIntegrations: OnCallIntegrationDTO[] | undefined | null,
+  resultFromConfig?: AlertManagerCortexConfig,
+  resultFromReadOnlyEndpoint?: Receiver[]
 ): ContactPointWithMetadata[] {
-  const contactPoints = result.alertmanager_config.receivers ?? [];
-
-  // compute the entire inherited tree before finding what notification policies are using a particular contact point
-  const fullyInheritedTree = computeInheritedTree(result?.alertmanager_config?.route ?? {});
-  const usedContactPoints = getUsedContactPoints(fullyInheritedTree);
-  const usedContactPointsByName = countBy(usedContactPoints);
+  let fullyInheritedTree: Route | undefined,
```

Comment:
I don't like let, but in this case i found more readable the code bellow using it.

Reply:
> What parts of the function definition prior to this change did you find confusing?

maybe I did not explain correctly. After the change, we need to assign different values to this const. We can use something like if you prefer.

```
const fullyInheritedTree = resultFromConfig && computeInheritedTree(resultFromConfig?.alertmanager_config?.route ?? {});
const usedContactPoints = resultFromConfig && fullyInheritedTree && getUsedContactPoints(fullyInheritedTree);
const usedContactPointsByName = resultFromConfig && countBy(usedContactPoints);
const contactPoints = resultFromConfig ? (resultFromConfig?.alertmanager_config.receivers ?? []) : (resultFromReadOnlyEndpoint ?? []);
```

Reply:
I don't think we should have to assign different values to them since `fullyInheritedTree` would just return `{}` (which is a valid route) when the alertmanager config is `undefined`. 

The rest of the variables here would continue to work with an empty route and `numberOfPolicies[contactPoint.name]` would always yield `undefined` – which is what we wanted :)

Reply:
good point. 
Done!